WcaOnRails/app/models/wcif.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Wcif < ApplicationRecord
+  belongs_to :competition
+
+  # See https://github.com/thewca/worldcubeassociation.org/wiki/wcif
+  def json
+    json_db = JSON.parse(super || "{}")
+    json_db.merge(
+      "formatVersion" => "1.0",
+      "id" => competition.id,
+    )
+  end
+
+  def json=(new_json)
+    json = self.json.merge(new_json)
+    super(JSON.generate(json))
+  end
+
+  def json_hash
```

Comment:
Aha! That's a reasonable approach to that problem, but I think it's premature to work on that now. Because...

1. it's easier to work on one thing at a time
2. depending on how we build the basics, we might pick a slightly - or very - different solution. I can see some potential problems with this one, and some potentially good alternatives.
3. I think this is more of a "nice to have" feature, rather than a must